spec/presenters/hyrax/iiif_manifest_presenter_spec.rb

```diff
@@ -0,0 +1,213 @@
+
+# rubocop:disable BracesAroundHashParameters maybe a rubocop bug re hash params?
+RSpec.describe Hyrax::IiifManifestPresenter do
+  subject(:presenter) { described_class.new(work) }
+  let(:work) { build(:monograph) }
+
+  describe 'manifest generation' do
+    let(:builder_service) { Hyrax::ManifestBuilderService.new }
+
+    it 'generates a IIIF presentation 2.0 manifest' do
+      expect(builder_service.manifest_for(presenter: presenter))
+        .to include('@context' => 'http://iiif.io/api/presentation/2/context.json')
+    end
+
+    context 'with file set and work members' do
+      let(:work) { create(:work_with_image_files) }
+
+      it 'generates a manifest with nested content' do
+        expect(builder_service.manifest_for(presenter: presenter)['sequences'].first['canvases'].count)
+          .to eq 2
+      end
+
+      context 'and an ability' do
+        let(:ability) { Ability.new(user) }
+        let(:user) { create(:user) }
+
+        before { presenter.ability = ability }
+
+        it 'excludes items the user cannot read' do
+          expect(builder_service.manifest_for(presenter: presenter))
+            .not_to have_key('sequences')
+        end
+
+        it 'includes items with read permissions' do
+          readable = FactoryBot.create(:file_set, :image, user: user)
+          work.members << readable
+          work.save
+
+          expect(builder_service.manifest_for(presenter: presenter)['sequences'].first['canvases'].count)
+            .to eq 1
+        end
+      end
+    end
+  end
+
+  describe Hyrax::IiifManifestPresenter::DisplayImagePresenter do
+    subject(:presenter) { described_class.new(solr_doc) }
+    let(:solr_doc) { SolrDocument.new(file_set.to_solr) }
+    let(:file_set) { create(:file_set, :image) }
+
+    describe '#display_image' do
+      it 'gives a IIIFManifest::DisplayImage' do
+        expect(presenter.display_image.to_json)
+          .to include 'fcr:versions%2Fversion1/full'
+      end
+
+      context 'with non-image file_set' do
+        let(:file_set) { create(:file_set) }
+
+        it 'returns nil' do
+          expect(presenter.display_image).to be_nil
+        end
+      end
+    end
+  end
+
+  describe '#description' do
+    it 'returns a string description of the object' do
+      expect(presenter.description).to be_a String
+    end
+  end
+
+  describe '#file_set_presenters' do
+    it 'is empty' do
+      expect(presenter.file_set_presenters).to be_empty
+    end
+
+    context 'when the work has file set members' do
+      let(:work) { create(:work_with_image_files) }
+
+      it 'gives presenters for the file sets' do
+        expect(presenter.file_set_presenters)
+          .to contain_exactly(*work.member_ids.map { |id| have_attributes(id: id) })
+      end
+
+      it 'gives DisplayImagePresenters' do
+        expect(presenter.file_set_presenters.map(&:display_image))
+          .to contain_exactly(an_instance_of(IIIFManifest::DisplayImage),
```

Comment:
This appears to be related to the above "magic" number 2. Wondering how to convey that in the spec.